add notification time report

mkangia opened the pull request:
https://trello.com/c/CKBqy6CG/252-gather-stats-to-check-for-time-take-for-nikshay-notifications

buddy @emord

mkangia committed:
add notification time report

proteusvacuum commented on custom/enikshay/management/commands/nikshay_registration_notification_time_report.py:
```diff
@@ -0,0 +1,73 @@
+                    'notification completed on': time_of_notification.strftime('%Y-%m-%d-%H:%M:%S'),
+                    'form to submission': (form_received_on - property_modified_on),
+                    'submission to notification': (time_of_notification - form_received_on),
+                    'enikshay id': episode_case.case_id
```
This isn't the enikshay_id. Normally, they would want the "readable ID" or the case_id of the person case.

mkangia replied:
This is for reference to quickly get to the case detail page. Renaming it to case id to avoid confusion.

proteusvacuum commented on corehq/ex-submodules/casexml/apps/case/util.py:
```diff
@@ -184,34 +184,37 @@ def iter_cases(case_ids, strip_history=False, wrap=True):
     for i, transactions in enumerate(actions):
         property_changed_info = property_changed_in_action(transactions, case_property_name)
         if property_changed_info and property_changed_info.new_value == value:
             # get the date that case_property changed
```
you might want to delete this comment from here

proteusvacuum commented on corehq/ex-submodules/casexml/apps/case/util.py:
```diff
@@ -184,34 +184,37 @@ def iter_cases(case_ids, strip_history=False, wrap=True):
-        for (modified_on, action) in update_actions:
-            if action:
-                property_changed = action.dynamic_properties.get(case_property_name)
+        for (modified_on, update_action, case_update) in update_actions:
```
it's confusing to me that this is called `action` when this list is created, but `case_update` here. Which is it?

mkangia replied:
oh yes.. this is very confusing. 
Improving.

mkangia committed:
remove redundant comment

mkangia committed:
avoid error in case of missing form for an update.

A case update, for ex: type 16 which is ERROR does not have a corresponding form submission. This fails when trying to extract case blocks. So return empty list in case there is no form.

mkangia committed:
better naming
action -> case transaction

mkangia committed:
Add task to notify report each day

proteusvacuum reviewed (approved):
Is there a reason you didn't make this into an actual report accessible on HQ?

mkangia commented:
> Is there a reason you didn't make this into an actual report accessible on HQ?

1. Initially intended to just this once but now thinking on receiving this for few more days and then remove it when and if not needed.
2. Also simply didn't know this could be made into an actual report on HQ

@proteusvacuum

proteusvacuum commented on custom/enikshay/management/commands/nikshay_registration_notification_time_report.py:
```diff
@@ -0,0 +1,92 @@
+            writer = csv.DictWriter(csvfile,
+                                    fieldnames=["nikshay id", "form finished on", "form submitted on",
+                                                "notification completed on", "form to submission",
+                                                "submission to notification", "enikshay id"])
```
I think this should say "case id" now

mkangia replied:
Ah! silly me.

mkangia committed:
fix issue with csv header [skip ci]
